pkg/lib/machine-info.js

```diff
@@ -193,3 +193,87 @@ export function udev_info(address) {
     }
     return pr;
 }
+
+const memoryRE = /^([ \w]+): (.*)/;
+
+// Process the dmidecode output and create a mapping of locator to DIMM properties
+function parseMemoryInfo(text) {
+    var info = {};
+    text.split("\n\n").map(paragraph => {
+        let locator = null;
+        let props = {};
+        paragraph = paragraph.trim();
+        if (!paragraph) {
+            return;
+        }
+        paragraph.split("\n").map(line => {
+            line = line.trim();
+            let match = line.match(memoryRE);
+            if (match) {
+                props[match[1]] = match[2];
+            }
+        });
+
+        locator = props["Locator"];
+        if (locator) {
+            info[locator] = props;
+        }
+    });
+    return processMemory(info);
+}
+
+// Select the useful properties to display
+
+function processMemory(info) {
+    let memoryArray = [];
+
+    for (let dimm in info) {
+        let memoryProperty = info[dimm];
+
+        let memorySize = memoryProperty["Size"];
+        if (memorySize.includes("MB")) {
+            let memorySizeValue = parseInt(memorySize, 10);
+            memorySize = memorySizeValue / 1024 + " GB";
+        }
+
+        let memoryTechnology = memoryProperty["Memory Technology"];
+        if (!memoryTechnology || memoryTechnology == "<OUT OF SPEC>") {
+            memoryTechnology = "Unknown";
+        }
+
+        let memoryRank = memoryProperty["Rank"];
+        if (memoryRank == 1) memoryRank = "Single Rank";
+        if (memoryRank == 2) memoryRank = "Dual Rank";
+
+        memoryArray.push({
+            locator: memoryProperty["Locator"],
+            technology: memoryTechnology,
+            type: memoryProperty["Type"],
+            size: memorySize,
+            remainingRatedWriteEndurance: memoryProperty["Memory Technology"] == "Intel persistent memory" ? "100%" : "Unknown",
```

Comment:
In case of persistent memory  (NVDIMMs), it takes tens of thousands of writes to wear of the memory module, so the value will be 100% but only for persistent memory modules